src/actions/AccountReferralActions.ts

```diff
@@ -330,7 +330,7 @@ export async function validatePromoCardsInner(dataStore: EdgeDataStore, cards: M
 async function saveAccountReferral(state: RootState): Promise<void> {
   const { account } = state.core
   const { accountReferral } = state.account
-  await account.disklet.setText(ACCOUNT_REFERRAL_FILE, JSON.stringify(accountReferral))
+  await account?.disklet?.setText(ACCOUNT_REFERRAL_FILE, JSON.stringify(accountReferral))
```

Comment:
These code paths shouldn't be activated if the state is invalid... What's causing these code paths to be reached before account is ready?

Reply:
account object gets cleared when the core is unmounted, likely due to webview getting killed

Reply:
If the `account` object is being set to the hack object, then that means we've logged out. If we've logged out, why are these functions being called? They shouldn't be running unless maybe this is caused by unresolved promises.. Eh, this is fine for now because this is just how things are with shared state and concurrency.